### PR TITLE
slideshow: add option to enable legacy presentation

### PR DESCRIPTION
**Pull request by hcvcastro**
Changes done.

### CODE REVIEW

**Comment by timar** [comment]
Experimental mode is not only for the new presentation engine, it is a general switch. I can imagine that currently we use it only for the new presentation engine, but it is a coincidence, not a rule. 
I thought we wanted to make the new presentation engine the default now, in the current release cycle. So probably it makes sense to allow to switch back to legacy SVG download mode, even in COOL 24.04.10.3.
Let's wait for the final decision, whether we can make the new presentation engine the default now.

**Comment by hcvcastro** [comment]
Oh I forgot to stage a change

**Comment by hcvcastro** [comment]
Yep, slideshow option is OK

File: browser/js/global.js
Comment (eszkadev) [inline browser/js/global.js:395]: can we follow the pattern: `slideshow` -> `slideshowEnabled`
or better `slideshow` -> `canvasSlideshowEnabled` or something similar to tell what exactly that option does (for someone unfamiliar)

File: coolwsd.xml.in
Comment (eszkadev) [inline coolwsd.xml.in:127]: 1. If we want to enable it by default (webgl) we have to remember this option in not present in the already created config files. That means we need to by default assume `true` if not present (to check in the C++ code).
2. Can we put that "feature flag" at the end of config where all other feature flags are?
3. Can we change name so it will explicitly say what it does without looking into description here (when reading code)? `slideshow` -> `canvasSlideshowEnabled` or similar.
4. Please add missing `type="bool"`
5. Description: `If true, WebGl presentation rendered on the client side is enabled, otherwise interactive SVG is used`

File: wsd/FileServer.cpp
Comment (eszkadev) [inline wsd/FileServer.cpp:1417]: 1. last parameter should be true - we want to enable it by default when config option is missing - see my comment in the config file
2. `%SLIDESHOW_ENABLED%` -> `%CANVAS_SLIDESHOW_ENABLED%`

File: wsd/DocumentBroker.cpp
Comment (eszkadev) [reply wsd/DocumentBroker.cpp:1403]: also logic is not preserved here:
We want to disable presentation when new slideshow is OFF and watermark text is SET (because svg presentation doesn't support watermarking but canvas one yes)
so it shoudl be `... || ( !canvasSlideshowEnabled && !watermarkText.empty() )`

**Comment by hcvcastro** [comment]
Changes done.

File: coolwsd.xml.in
Comment (eszkadev) [inline coolwsd.xml.in:127]: can we move it at the end of file? now it is between logging options. at the end there are other on/off feature flags

**Review by eszkadev — APPROVED** [review]
thanks, I didn't test but code looks ok now :)

**Pull request merged**
